CONTRIBUTING_JS.md

```diff
@@ -299,23 +267,23 @@ const API = require('ipfs-api')
 
 There are two possibilities: either it didn’t work out for us, or we don’t know about it. If you think we might have missed it please tell us, but please believe us if we say we tried and it didn’t work for us.
 
-#### Why not use simple npm scripts instead of gulp?
-
-Gulp is not a hard dependency. It’s just a simple way to structure our tasks at the moment. Usually projects only depend on the aegir binaries completely hiding the fact that we are using gulp under the hood. So we are free if we want to switch it out without any issues. We all enjoy npm scripts, and are using them to call the aegir binaries, but there is no nice way of sharing them yet.
-
 #### Where are all the semicolons?
 
 Our linting rules are compatible with [standard](https://github.com/feross/standard), which has many examples on documentation on this. Please go there and read it if you're still curious.
 
-#### Why are you bothering with ES2015 and all this build setup?
+#### Why are you bothering with TypeScript and all this build setup?
+
+We have a large number of modules that integrate together and we've found that the only way to do this at scale without increasing friction to burning point is to have a type system.
+
+At the time of writing a type system based on TypeScript [may be coming to JavaScript](https://github.com/tc39/proposal-type-annotations) so we're only a little ahead of the curve on that front.
 
-We want to see the web move forward, and some of us enjoy writing their JavaScript with things like `const` and arrow functions.
+We've also found that having types means there's less magic in our codebases since it becomes harder to do use the extreme ends of JavaScript's flexibility which then makes everything easier to follow, lowering cognitive overhead and maintenance burden.
```

Comment:
```suggestion
We've also found that having types means there's less magic in our codebases since it becomes harder to use the extreme ends of JavaScript's flexibility, making everything easier to follow, and lowering cognitive overhead and maintenance burden.
```